### ISSUE TITLE
Digito verificador do nosso numero no SANTANDER

### ISSUE THREAD

**Issue by fabioginzel** [fredroo/openboleto#32]
Segundo a documentação do santander, o nosso número possuí um digito verificador calculado com módulo 11, no fonte ele esta concatenando o sequencial com o "carteiradv" não seria melhor calcular o dígito?

**Comment by Rctnet** [comment]
Boa noite @fabioginzel , seria sim, olhei a documentação e você esta correto, poderia fazer essa alteração e homologar junto banco? assim podemos criar um mapa de tudo que estiver homologado

**Comment by fabioginzel** [comment]
Sim, pode deixar vou homologar o boleto do santander e te aviso..
Vou fazer essa alteração do calculo do digito e mando um pull request

**Comment by aledz7** [comment]
Conseguiu homologar? Estou travado também nesse digito verificador.

**Comment by fabioginzel** [comment]
Ainda não consegui enviar para homologação, muitos projetos aqui.. mas o plano é até setembro estar com isso funcionando... mas pelos testes que fiz comparando com minha outra ferramenta já homologada esta OK..
POsso mandar um pull request se quiser @aledz7

**Comment by Rctnet** [comment]
Bom dia a todos, @fabioginzel , manda uma PR com o que tem ai, já vai ser útil.

**Comment by fabioginzel** [comment]
Pronto.. só aceitar

**Comment by aledz7** [comment]
Oi @fabioginzel . Seria de muita ajuda! 
Mas tenho a mínima ideia de como faz para usar o PR. Sou iniciante no git. Criei o diretório mas não consegui publicar. Diz que preciso fazer upgrade pra poder publicar algo: https://github.com/aledz7/open_boleto

O pessoal do santander respondeu: Críticas Boletos: 
 
Código de Barras: 
Como Está: 0033997580000001200090288511200000000000580101
Deve ser:    003399758000000120009022445600000000000510101
 
Linha digitavel: 
 
Como está: 03399.02884 51120.000008 00000.580100 9 75800000012000
Deve ser:   03399.02444 55620.000008 00000.510100 9 75800000012000

Baixei esse script do seu repositório.

**Comment by Rctnet** [comment]
Relaxa @aledz7 , o @fabioginzel  esta resolvendo um bugzinho que deu no traviz, assim que ele resolver eu aceito a PR (sou adm do projeto) ai voce pode baixar normalmente na home do projeto

**Comment by fabioginzel** [comment]
Acabei de chegar do almoço.. rs.. ja faço isso..

**Comment by Rctnet** [comment]
O @fabioginzel  acabou de liberar o update, já está disponível na home do projeto

**Comment by aledz7** [comment]
Bacana. Eu to com uma dúvida.
O que eu coloco nesse campo novo? 'carteiraDv' => 1,

?

**Comment by fabioginzel** [comment]
Nada.. esse campo não existe mais.. porém precisa esperar o @fredroo arrumar o packagist..
vc esta vendo a versão antiga da biblioteca ainda..
Outra opção é fazer download manual da biblioteca

**Comment by fredroo** [comment]
atualizado

**Comment by fredroo** [comment]
@Rctnet qual é o seu usuário no packagist? para eu te adicionar nesse projeto lá

**Comment by Rctnet** [comment]
RogerioCastro

**Comment by fredroo** [comment]
@Rctnet verifica se apareceu ai para você as permissões

**Comment by Rctnet** [comment]
sim deu certo

**Comment by aledz7** [comment]
Bacana. Baixei de novo. Mas em samples ainda aparece o 'carteiraDv' => 1, no santander.php
Ignora este parâmetro então?

**Comment by fabioginzel** [comment]
Sim.. eu não tinha mexido em "samples".. Seria legal vc acertar o samples e enviar o codigo para que os adms aceitem

**Comment by aledz7** [comment]
Blz. Faço a alteração e envio após homologação.
Mas não deu certo ainda. O código de barras ficou o mesmo.
É para eu baixar desse certo? https://github.com/fabioginzel/openboleto

**Comment by fabioginzel** [comment]
Não... agora ja pode usar o composer update normalmente...

**Comment by aledz7** [comment]
Dei update. Deu:
Nothing to install or update

**Comment by aledz7** [comment]
Isso que estou achando estranho.

**Comment by fabioginzel** [comment]
Verifique se na classe do santander ainda existe o campo carteiraDV

Se existir, então vc ta olhando a versão antiga... apaga seu vendor e da composer update novamente

**Comment by aledz7** [comment]
O pessoal da santander respondeu:

Críticas Boletos: 
 
Código de Barras: 
Como Está: 0033997580000001200090288511200000000000580101
Deve ser:    003399758000000120009022445600000000000510101
 
Linha digitavel: 
 
Como está: 03399.02884 51120.000008 00000.580100 9 75800000012000
Deve ser:   03399.02444 55620.000008 00000.510100 9 75800000012000

Sinceramente sei nem pra onde ir rs.

**Comment by fabioginzel** [comment]
@aledz7 algum retorno?

**Comment by aledz7** [comment]
Sim. Falaram que não houve alterações. Mas verifiquei e houve sim. Rebati dizendo que o código de barras era outro e etc. Para eles verificarem de novo.

**Comment by aledz7** [comment]
Segue novas críticas do boleto santander:

Boleto: 

Código de Barras incorreto
Como está: 03393758000000120009288511200000000005820101
Deve ser:    03393758000000120009022445600000000005820101  

Linha digitavel incorreta. 
Como está: 03399.28855 11200.000005 00058.201013 3 75800000012000 
Deve ser:    03399.02245 45600.000005 00058.201013 3 75800000012000 

Poderiam verificar? Aguardo retorno.

**Comment by fabioginzel** [comment]
Consigo ver isso só mes que vem.. Se puder olhar o fonte e descobrir o problema agradeço

**Comment by aledz7** [comment]
Rapaz. Eu num tenho a mínimo ideia de por onde começar.
:(

**Comment by fabioginzel** [comment]
:( rs 
Ta.. Calma.. vou ver isso pra vc no domingo.. ta?
Depois me paga uma breja...

**Comment by aledz7** [comment]
kkk. Combinado então!
Obrigado mesmo!

**Comment by aledz7** [comment]
@fabioginzel . Tu conseguiu dar uma olhada?

**Comment by fabioginzel** [comment]
Desculpa.. acabei viajando...
Me passa os dados desse boleto que saiu errado.. (conta,agencia,valor,dtvenc.. etc)

**Comment by aledz7** [comment]
Opa.
Coloquei assim:

    'agencia' => 4391, // Até 4 dígitos
    'carteira' => 101, // 101, 102 ou 201
    'conta' => 013002770, // Código do cedente: Até 7 dígitos
     // IOS – Seguradoras (Se 7% informar 7. Limitado a 9%)
     // Demais clientes usar 0 (zero)
    'ios' => '0', // Apenas para o Santander

    // Parâmetros recomendáveis
    //'logoPath' => 'http://empresa.com.br/logo.jpg', // Logo da sua empresa
    'contaDv' => '0',
    'agenciaDv' => '5',
	'carteiraDv' => 1,

**Comment by fabioginzel** [comment]
Faltou o resto dos daos para eu poder simular..  Valor, sequencia.. datas... etc.. me passa os dados completos

**Comment by aledz7** [comment]
Seria isso?

$sacado = new Agente($rs_clientes->nome, $rs_clientes->cnpj, $rs_clientes->endereco, $rs_clientes->cep, $rs_clientes->nomeCidade, $rs_clientes->nomeEstado);
$cedente = new Agente('SUPERCORR ADMINISTRADORA E CORRETORA', '19.115.270/0001-76', 'Q SRTVS BLOCO K SALA 428 EDIF EMBASSY TOWER	', '70.340-908', 'Brasília', 'DF');

$boleto = new Santander(array(
    // Parâmetros obrigatórios
    'dataVencimento' => new DateTime($rs_contas->data_vencimento),
    'valor' => $rs_contas->valor_total,
    'sequencial' => str_pad($rs_contas->id, 12, "0", STR_PAD_LEFT), // Até 13 dígitos
    'sacado' => $sacado,
    'cedente' => $cedente,
    'agencia' => 4391, // Até 4 dígitos
    'carteira' => 101, // 101, 102 ou 201
    'conta' => 013002770, // Código do cedente: Até 7 dígitos
     // IOS – Seguradoras (Se 7% informar 7. Limitado a 9%)
     // Demais clientes usar 0 (zero)
    'ios' => '0', // Apenas para o Santander

    // Parâmetros recomendáveis
    //'logoPath' => 'http://empresa.com.br/logo.jpg', // Logo da sua empresa
    'contaDv' => '0',
    'agenciaDv' => '5',
	'carteiraDv' => 1,
    'descricaoDemonstrativo' => array( // Até 5
        'Serviços de Corretagem de Seguros',
    ),
    'instrucoes' => array( // Até 8
        'Após o vencimento cobrar 2% de mora e 0.33% de juros ao dia.',
    ),

    // Parâmetros opcionais
    //'resourcePath' => '../resources',
    //'moeda' => Santander::MOEDA_REAL,
    //'dataDocumento' => new DateTime(),
    //'dataProcessamento' => new DateTime(),
    //'contraApresentacao' => true,
    //'pagamentoMinimo' => 23.00,
    //'aceite' => 'N',
    //'especieDoc' => 'ABC',
    //'numeroDocumento' => '123.456.789',
    //'usoBanco' => 'Uso banco',
    //'layout' => 'layout.phtml',
    //'logoPath' => 'http://boletophp.com.br/img/opensource-55x48-t.png',
    //'sacadorAvalista' => new Agente('Antônio da Silva', '02.123.123/0001-11'),
    //'descontosAbatimentos' => 123.12,
    //'moraMulta' => 123.12,
    //'outrasDeducoes' => 123.12,
    //'outrosAcrescimos' => 123.12,
    //'valorCobrado' => 123.12,
    //'valorUnitario' => 123.12,
    //'quantidade' => 1,
));
